tests/Aspire.MongoDB.Driver.Tests/Aspire.MongoDB.Driver.Tests.csproj

```diff
@@ -0,0 +1,13 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>$(NetCurrent)</TargetFramework>
+    <NoWarn>CS8002;CS1591</NoWarn>
```

Comment:
Because the official MongoDB.Driver package doesn't have a strongly name and it causes compilation errors.

https://jira.mongodb.org/browse/CSHARP-142

Reply:
I've updated with the suggestion, can we close this one?

Reply:
@jkotas @dsplaisted - are there any drawbacks/limitations/issues in .NET 8+ with using weak named assemblies?

One that @ericstj told me was:

CoreCLR will load any version of a weak named assembly. So even if you reference `v2`, if `v1` happens to be lying around the runtime will load it and you'll get crazy errors later.

Our guidance https://learn.microsoft.com/en-us/dotnet/standard/library-guidance/strong-naming is that people consider strong naming their libraries. Although adding one is a binary breaking change.

Reply:
> CoreCLR will load any version of a weak named assembly. So even if you reference `v2`, if `v1` happens to be lying around the runtime will load it and you'll get crazy errors later.

I didn't know that was the case, I'm kind of surprised by that.

Reply:
> CoreCLR will load any version of a weak named assembly. So even if you reference v2, if v1 happens to be lying around the runtime will load it and you'll get crazy errors later.

This questionable behavior is specific to .NET Framework. 

.NET Core/5+ assembly loader respects assembly version by default, irrespective of strong name. It requires assembly definition version >= assembly reference version. 

(The default behavior can be overridden by a custom assembly resolver. The assembly loader does not validate versions of assemblies returned by custom assembly resolver.)

Reply:
Thanks, @jkotas. Since Aspire Components only target `net8.0`+ there shouldn't be any major issues here with referencing a weak named assembly. I'd still think we would recommend the underlying libraries be strong named, but it sounds like it shouldn't be a blocker.

Reply:
https://learn.microsoft.com/en-us/dotnet/standard/library-guidance/strong-naming#create-strong-named-net-libraries

says

> You should strong name your open-source .NET libraries.

Reply:
I'm hesitant to remove strong naming on all our libraries. I really don't want Aspire to be the testing ground here. Especially if getting it wrong means we need to make a breaking change in the future.

I could imagine a solution here being MongoDB Aspire Component isn't strong named because the underlying MongoDB library isn't strong named. That seems like an OK strategy.

@terrajobst @DamianEdwards @davidfowl @joperezr any thoughts here?

Reply:
Agree with concerns about Aspire being the canary here. Feels more like it should be a more centrally coordinated change across the suite of .NET products than something we just start doing. Suppressing the warning (given it's just a compiler-level thing) seems fine for us.

Reply:
> we update the docs to make that clearer

https://github.com/dotnet/docs/pull/38537